presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -1379,8 +1380,14 @@ private static HiveStorageFormat extractHiveStorageFormat(Table table)
 
     private static Function<HiveColumnHandle, ColumnMetadata> columnMetadataGetter(Table table, TypeManager typeManager)
     {
+        Set<String> partitionKeyNames = table.getPartitionKeys().stream().map(FieldSchema::getName).collect(Collectors.toSet());
+        List<FieldSchema> tableColumns = table.getSd().getCols();
+        if (tableColumns.stream().anyMatch(field -> partitionKeyNames.contains(field.getName()))) {
+            throw new PrestoException(HIVE_INVALID_METADATA, "Partition column cannot also be a table column");
```

Comment:
This is a weird message. From Presto's (and Presto users) point of view, partition columns *are* table columns.

What's the issue we're trying to deal with here? What fails if a column name appears in both?